Lib/test/test_asyncio/test_tasks.py

```diff
@@ -1009,16 +1009,63 @@ def gen():
         self.assertEqual(res, "ok")
 
     def test_wait_for_cancellation_race_condition(self):
+        def gen():
+            yield 0.1
+            yield 0.1
+            yield 0.1
+            yield 0.1
+
+        loop = self.new_test_loop(gen)
+
+        fut = self.new_future(loop)
+        # Test that if task is cancelled at the same time the future
+        # completes, the result is still returned.
+        loop.call_later(0.1, fut.set_result, "ok")
+        task = loop.create_task(asyncio.wait_for(fut, timeout=1))
+        loop.call_later(0.1, task.cancel)
+        res = loop.run_until_complete(task)
+        self.assertEqual(res, "ok")
```

Comment:
Putting this test back in, with a comment. It does manage to trigger the race condition aaliddel mentioned.